MIGRATION.md

```diff
@@ -147,23 +204,80 @@ There is no codemod for these changes, migrate manually through search and repla
 
 > Generally, avoid using typography size tokens. Instead, use typography components directly.
 
-### Modal improvements
+### Consolidated modals and popovers
 
-(...)
+The `Modal` and `Popover` components were refactored to consolidate their APIs and to improve their accessibility.
 
-### Accessibility
+#### Modals
+
+The `Modal`, `ModalWrapper`, `ModalHeader`, `ModalFooter`, `ModalContext`, and `ModalConsumer` components are no longer exported. Instead, use the `useModal` hook to render modals instead.
+
+Modals in Circuit v3 are accessible by default, have a streamlined UI and behavior (dimensions, click outside, etc.) and handle edge cases like modal and popover stacking.
+
+Refer to [the Modal stories](https://circuit.sumup.com/?path=/story/components-modal--base) for usage examples.
+
+#### Popovers
+
+The `Popover` component was rebuilt in Circuit v3. It now uses [Popper v2](https://popper.js.org/) under the hood and comes with a refreshed component API.
+
+More importantly, the Circuit v3 Popover is opinionated when it comes to its usage pattern. It no longer accepts custom children, unlike v2, but rather a list of actions (links and/or buttons).
+
+To migrate, separate the popovers in your application into two types.
+
+Start by migrating any popover resembling a dropdown menu to the new Circuit v3 Popover. Refer to [the Popover story](https://circuit.sumup.com/?path=/story/components-popover--base) for a usage example.
+
+Most of the remaining popovers with custom children should gradually be migrated to use different UI patterns (for example modals). In the meantime, we recommend creating a local copy of the Circuit v3 Popover in your application that accepts custom children instead of an array of actions. You'll find [an example](https://github.com/sumup/ze-dashboard/blob/master/src/components/Popover/Popover.tsx) in the SumUp merchant dashboard (private repository).
```

Comment:
Again, not happy about the link to internal code, though I can't think of an alternative. I like the `(private repository)` label.

Reply:
Agreed, but should we leave it in this case if we don't have an alternative? I think it could be helpful to people who have access to see how the local component was implemented.

Reply:
Yes, let's leave it with the label.